src/python/pants/java/nailgun_executor.py

```diff
@@ -228,7 +228,10 @@ def _await_socket(self, timeout):
           accumulated_stdout += line
 
   def _create_ngclient(self, port, stdout, stderr, stdin):
-    return NailgunClient(port=port, ins=stdin, out=stdout, err=stderr)
+    kwargs = {"ins": stdin, "out": stdout, "err": stderr}
```

Comment:
Why not just fix this in NailgunClient so you know it's fixed for all code paths? Change the defaulting their in the constructor params to None and then in the constructor body if None, use the intended default.

Reply:
Changed.

Generally I think it's helpful to keep the default value the actual default you plan to use, rather than `None`, so that things like Intellisense are more helpful. But, we use that `None` pattern quite a bit in this code so I'm going with convention because it brings the benefits you mention.